File: client/bcephal-client-model/Expressions/ExpressionValue.cs
using Bcephal.Models.Base;
using Bcephal.Models.Filters;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Expressions
{
    public class ExpressionValue
    {

		public ExpressionValueType? Type { get; set; }

		public decimal? DecimalValue { get; set; }

		public PeriodValue PeriodValue { get; set; }

		public string StringValue { get; set; }

		public long? IdValue { get; set; }

		public long? DimensionId { get; set; }

		public DimensionType? DimensionType { get; set; }



        [JsonIgnore]
        public  bool IsColumn { get { return Type.HasValue && Type.Value.IsColumn(); } }

        [JsonIgnore]
        public bool IsParameter { get { return Type.HasValue && Type.Value.IsParameter(); } }

        [JsonIgnore]
        public bool IsSpot { get { return Type.HasValue && Type.Value.IsSpot(); } }

        [JsonIgnore]
        public bool IsLoop { get { return Type.HasValue && Type.Value.IsLoop(); } }

        [JsonIgnore]
        public bool IsNull { get { return Type.HasValue && Type.Value.IsNull(); } }

        [JsonIgnore]
        public bool IsNo { get { return Type.HasValue && Type.Value.IsNo(); } }



        [JsonIgnore]
        public bool IsAttribute { get { return DimensionType.HasValue && DimensionType.Value.IsAttribute(); } }

        [JsonIgnore]
        public bool IsMeasure { get { return DimensionType.HasValue && DimensionType.Value.IsMeasure(); } }

        [JsonIgnore]
        public bool IsPeriod { get { return DimensionType.HasValue && DimensionType.Value.IsPeriod(); } }


    }
}

File: server/bcephal-base/src/main/java/com/moriset/bcephal/domain/expression/ExpressionValue.java
/**
 * 
 */
package com.moriset.bcephal.domain.expression;

import java.math.BigDecimal;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.moriset.bcephal.domain.dimension.DimensionType;
import com.moriset.bcephal.domain.filters.PeriodValue;

import lombok.Data;

/**
 * @author Joseph Wambo
 *
 */
@Data
public class ExpressionValue {

	private ExpressionValueType type;

	private BigDecimal decimalValue;

	private PeriodValue periodValue;

	private String stringValue;

	private Long idValue;

	private Long dimensionId;

	private DimensionType dimensionType;

	@JsonIgnore
	public boolean isColumn() {
		return type == ExpressionValueType.COLUMN;
	}

	@JsonIgnore
	public boolean isLoop() {
		return type == ExpressionValueType.LOOP;
	}

	@JsonIgnore
	public boolean isSpot() {
		return type == ExpressionValueType.SPOT;
	}

	@JsonIgnore
	public boolean isNoCondition() {
		return type == ExpressionValueType.NO;
	}

	@JsonIgnore
	public boolean isNull() {
		return type == ExpressionValueType.NULL;
	}

	@JsonIgnore
	public Object getValue() {
		if (isNoCondition()) {
			return null;
		}
		if (isNull()) {
			return null;
		}
		if (dimensionType != null) {
			if (dimensionType.isAttribute()) {
				return stringValue;
			}
			if (dimensionType.isMeasure()) {
				return decimalValue;
			}
			if (dimensionType.isPeriod()) {
				return periodValue.buildDynamicDate();
			}
		}
		return null;
	}

}
